File: src/Quest/Shared/Common/Model/Question.cs
using System;
using Newtonsoft.Json;
using System.Collections.Generic;

namespace Common.Model
{
    public class Question: DocumentEntity
    {
        public Question() { }
        public string Value { get; set; }
        public int DifficultLevel { get; set; }
        public string QuestionTypeCode { get; set; }
        public string AnswerTypeCode { get; set; }
        public string CorrectAnswer { get; set; }
        public AnswerOptions Options { get; set; }
        public List<Category> Categories { get; set; }
        public string Tags { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public string LastModifiedBy { get; set; }
        public DateTime LastModifedOn { get; set; }
        public override string ToString()
        {
            return JsonConvert.SerializeObject(this);
        }

        public override bool Equals(object obj)
        {
            var otherQuestion = obj as Question;
            if (otherQuestion == null)
                return false;

            return (Id == otherQuestion.Id
                && Value == otherQuestion.Value
                && CorrectAnswer == otherQuestion.CorrectAnswer
                && AnswerTypeCode == otherQuestion.AnswerTypeCode
                && QuestionTypeCode == otherQuestion.QuestionTypeCode
                && Tags == otherQuestion.Tags
                && DifficultLevel == otherQuestion.DifficultLevel);
        }

        public override int GetHashCode()
        {
            return base.GetHashCode();
        }
    }

    public class AnswerOptions
    {
        public AnswerOptions() { }
        public int Option { get; set; }
        public string Code { get; set; }
        public string Answer { get; set; }
        public bool IsCorrect { get; set; }
    }
}

File: src/App/Android/app/src/main/java/com/example/quest/com/example/quest/models/Question.java
package com.example.quest.com.example.quest.models;

import java.util.Date;

public class Question {
    String id;
    String value;
    String answerTypeCode;
    String level;
    String questionTypeCode;
    //Category[] categories;
    Date lastModifiedOn;
    String lastModifiedBy;
    String[] tags;
    Boolean isPrivate;
    String createdBy;
    String correctAnswer;

    public Question(String id, String value, String answer, String level) {
        this.id = id;
        this.value = value;
        this.correctAnswer = answer;
        this.level = level;
    }

    public String getId() { return this.id; }

    public String getQuestion() {
        return this.value;
    }

    public String getAnswer() {
        return this.correctAnswer;
    }

    public String getLevel(){
        return this.level;
    }
}